subgraphs/uniswap-v3/src/common/entities/pool.ts

```diff
@@ -0,0 +1,208 @@
+import { Pool as PoolTemplate } from "../../../generated/templates";
+
+// Create a liquidity pool from pairCreated event.
+export function createLiquidityPool(
+  event: ethereum.Event,
+  poolAddress: Address,
+  token0Address: Address,
+  token1Address: Address,
+  fees: i32
+): void {
+  const protocol = getOrCreateProtocol();
+
+  // create the tokens and tokentracker
+  const token0 = getOrCreateToken(event, token0Address);
+  const token1 = getOrCreateToken(event, token1Address);
+
+  updateTokenWhitelists(token0, token1, poolAddress);
+
+  const pool = new LiquidityPool(poolAddress);
+  const poolAmounts = new _LiquidityPoolAmount(poolAddress);
+
+  pool.protocol = protocol.id;
+  pool.name =
+    protocol.name +
+    " " +
+    token0.name +
+    "/" +
+    token1.name +
+    " " +
+    convertFeeToPercent(fees).toString() +
+    "%";
+  pool.symbol = token0.name + "/" + token1.name;
+  pool.inputTokens = [token0.id, token1.id];
+  pool.inputTokenWeights = [BIGDECIMAL_FIFTY, BIGDECIMAL_FIFTY];
+
+  pool.fees = createPoolFees(poolAddress, fees);
+  pool.isSingleSided = false;
+  pool.createdTimestamp = event.block.timestamp;
+  pool.createdBlockNumber = event.block.number;
+
+  pool.liquidityToken = null;
+  pool.liquidityTokenType = TokenType.MULTIPLE;
+
+  pool.totalValueLockedUSD = BIGDECIMAL_ZERO;
+  pool.totalLiquidity = BIGINT_ZERO;
+  pool.totalLiquidityUSD = BIGDECIMAL_ZERO;
+  pool.activeLiquidity = BIGINT_ZERO;
+  pool.activeLiquidityUSD = BIGDECIMAL_ZERO;
+
+  pool.inputTokenBalances = [BIGINT_ZERO, BIGINT_ZERO];
+  pool.inputTokenBalancesUSD = [BIGDECIMAL_ZERO, BIGDECIMAL_ZERO];
+
+  pool.cumulativeTotalVolumeUSD = BIGDECIMAL_ZERO;
+  pool.cumulativeVolumeTokenAmounts = [BIGINT_ZERO, BIGINT_ZERO];
+  pool.cumulativeVolumesUSD = [BIGDECIMAL_ZERO, BIGDECIMAL_ZERO];
+
+  pool.cumulativeSupplySideRevenueUSD = BIGDECIMAL_ZERO;
+  pool.cumulativeProtocolSideRevenueUSD = BIGDECIMAL_ZERO;
+  pool.cumulativeTotalRevenueUSD = BIGDECIMAL_ZERO;
+
+  pool.uncollectedProtocolSideTokenAmounts = [BIGINT_ZERO, BIGINT_ZERO];
+  pool.uncollectedProtocolSideValuesUSD = [BIGDECIMAL_ZERO, BIGDECIMAL_ZERO];
+  pool.uncollectedSupplySideTokenAmounts = [BIGINT_ZERO, BIGINT_ZERO];
+  pool.uncollectedSupplySideValuesUSD = [BIGDECIMAL_ZERO, BIGDECIMAL_ZERO];
+
+  pool.positionCount = INT_ZERO;
+  pool.openPositionCount = INT_ZERO;
+  pool.closedPositionCount = INT_ZERO;
+
+  pool._totalAmountWithdrawn = [BIGINT_ZERO, BIGINT_ZERO];
+  pool._totalAmountCollected = [BIGINT_ZERO, BIGINT_ZERO];
+  pool._totalAmountEarned = [BIGINT_ZERO, BIGINT_ZERO];
+  pool._mostRecentSnapshotsDayID = INT_ZERO;
+  pool._mostRecentSnapshotsHourID = INT_ZERO;
+
+  poolAmounts.inputTokens = [token0.id, token1.id];
+  poolAmounts.inputTokenBalances = [BIGDECIMAL_ZERO, BIGDECIMAL_ZERO];
+  poolAmounts.tokenPrices = [BIGDECIMAL_ZERO, BIGDECIMAL_ZERO];
+
+  // Used to track the number of deposits in a liquidity pool
+  const poolDeposits = new _HelperStore(poolAddress);
```

Comment:
Oh, I like this 🤔 
Why did you decide to only use it for deposits

Reply:
This has been used in the pricing mechanism. But I agree that it would make sense to start tracking swap, deposit, and  withdraw counts at the pool level. Do you think I should add that in now ?

Reply:
Actually, the opposite. It might make sense to put those counters away from the pool entity because every change to them results in a whole copy of the Pool, increasing the DB and index size. So having values that change often in their dedicated store would make the `Pool` table more lightweight.

Let’s not add it for now, but we can keep it in mind for new schemas / subgraphs. Specially for the ones where we expect a lot of `Pools` to exist.